device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -745,7 +773,18 @@ DeviceIO getDeviceIO()
      */
     void setDeviceIO(DeviceIO deviceIO)
     {
+        // deviceIO may be set to null in the case when a device client was multiplexing and was unregistered
         this.deviceIO = deviceIO;
+
+        // Since connection status callbacks can be registered before associating a device client with a multiplexing client, the callback and its
+        // context also need to be registered when the device IO is set.
+        if (this.deviceIO != null && this.connectionStatusChangeCallback != null)
+        {
+            this.deviceIO.registerConnectionStatusChangeCallback(
```

Comment:
Q - is this called only for multiplexed devices?

Reply:
Yes, but both the old TransportClient and the new MultiplexingClient call into this. In both cases, they set the deviceIO of each device client to be the same deviceIO instance. That way they all share the same connection, worker threads, etc.